### PR TITLE
add qa project for running ILM tests against security

### PR DESCRIPTION
**Pull request by talevy**
add QA project for running ILM tests against security
    
This is a bare-bones skeleton for running existing yaml tests with security enabled. Additional tests which test users and roles should follow

### CODE REVIEW

File: x-pack/qa/smoke-test-index-lifecycle-with-security/build.gradle [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+apply plugin: 'elasticsearch.rest-test'
+
+dependencies {
+  testCompile project(path: xpackProject('plugin').path, configuration: 'testArtifacts')
+}
+
+// bring in machine learning rest test suite
+task copyILMRestTests(type: Copy) {
+  into project.sourceSets.test.output.resourcesDir
+  from project(xpackProject('plugin').path).sourceSets.test.resources.srcDirs
```
Comment (atorok) [inline x-pack/qa/smoke-test-index-lifecycle-with-security/build.gradle]: I realize we use this elsewhere, but was wondering if there's really a reason for it ? `xpackProject('plugin')` already returns a `Project` why wrap it in another call ? The additional `project(.path)` seems unnecessary.

File: x-pack/qa/smoke-test-index-lifecycle-with-security/build.gradle [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  setupCommand 'setupDummyUser',
+               'bin/elasticsearch-users',
+               'useradd', System.getProperty('tests.rest.cluster.username', 'test_user'),
+               '-p', System.getProperty('tests.rest.cluster.password', 'x-pack-test-password'),
+               '-r', 'superuser'
+  waitCondition = { node, ant ->
+    File tmpFile = new File(node.cwd, 'wait.success')
+    ant.get(src: "http://${node.httpUri()}/_cluster/health?wait_for_nodes=>=${numNodes}&wait_for_status=yellow",
+            dest: tmpFile.toString(),
+            username: 'test_user',
```
Comment (atorok) [inline x-pack/qa/smoke-test-index-lifecycle-with-security/build.gradle]: Would try to get this from the system properties as well. 
Maybe extract these into variables to remove repetition.
I'm assuming the properties are there so they can be used with external clusters  for which the wait conditions are not called, but that's an implementation detail  and looks odd not to consider the properties in here.

**Review by alpar-t — APPROVED** [review]
LGTM, just some nits

**Review by colings86 — APPROVED** [review]
LGTM2

**Comment by talevy** [comment]
thanks for the review @colings86 and @atorok. Your suggestions were helpful. I am too quick to turn a blind eye when it comes to gradle/groovy configuration

**Review by alpar-t — APPROVED** [review]

**Comment by talevy** [comment]
test this please

**Pull request merged**